src/test/scala/org/neo4j/spark/DataSourceReaderTSE.scala

```diff
@@ -330,7 +327,259 @@ class DataSourceReaderTSE extends SparkConnectorScalaBaseTSE {
+      (p3:Person {name: 'John Butler'})
+     """)
+
+    val result = df.where("name LIKE 'John%'").collectAsList()
+
+    assertEquals(3, result.size())
+  }
+
+  @Test
+  def testReadNodeWithEndsWith(): Unit = {
+    val df: DataFrame = initTest(s"""
+     CREATE (p1:Person {name: 'John Mayer'}),
+      (p2:Person {name: 'John Scofield'}),
+      (p3:Person {name: 'John Butler'})
+     """)
+
+    val result = df.where("name LIKE '%Scofield'").collectAsList()
+
+    assertEquals(1, result.size())
+  }
+
+  @Test
+  def testReadNodeWithContains(): Unit = {
+    val df: DataFrame = initTest(s"""
+     CREATE (p1:Person {name: 'John Mayer'}),
+      (p2:Person {name: 'John Scofield'}),
+      (p3:Person {name: 'John Butler'})
+     """)
+
+    val result = df.where("name LIKE '%ay%'").collectAsList()
+
+    assertEquals(1, result.size())
+  }
+
+  @Test
+  def testRelFiltersWithMap(): Unit = {
+    val fixtureQuery: String =
+      """UNWIND range(1,100) as id
+        |CREATE (p:Person {id:id,ids:[id,id]}) WITH collect(p) as people
+        |UNWIND people as p1
+        |UNWIND range(1,10) as friend
+        |WITH p1, people[(p1.id + friend) % size(people)] as p2
+        |CREATE (p1)-[:KNOWS]->(p2)
+        |RETURN *
+    """.stripMargin
+
+    SparkConnectorScalaSuiteIT.session()
+      .writeTransaction(
+        new TransactionWork[ResultSummary] {
+          override def execute(tx: Transaction): ResultSummary = tx.run(fixtureQuery).consume()
+        })
+
+    val df = ss.read.format(classOf[DataSource].getName)
+      .option("url", SparkConnectorScalaSuiteIT.server.getBoltUrl)
+      .option("relationship", "KNOWS")
+      .option("relationship.source.labels", "Person")
+      .option("relationship.target.labels", "Person")
+      .load()
+
+    assertEquals(1, df.filter("`<source>`.`id` = '10' AND `<target>`.`id` = '1'").collectAsList().size())
+  }
+
+  @Test
+  def testRelFiltersWithoutMap(): Unit = {
+    val fixtureQuery: String =
+      """UNWIND range(1,100) as id
+        |CREATE (p:Person {id:id,ids:[id,id]}) WITH collect(p) as people
+        |UNWIND people as p1
+        |UNWIND range(1,10) as friend
+        |WITH p1, people[(p1.id + friend) % size(people)] as p2
+        |CREATE (p1)-[:KNOWS]->(p2)
+        |RETURN *
+    """.stripMargin
+
+    SparkConnectorScalaSuiteIT.session()
+      .writeTransaction(
+        new TransactionWork[ResultSummary] {
+          override def execute(tx: Transaction): ResultSummary = tx.run(fixtureQuery).consume()
+        })
+
+    val df = ss.read.format(classOf[DataSource].getName)
+      .option("url", SparkConnectorScalaSuiteIT.server.getBoltUrl)
+      .option("relationship", "KNOWS")
+      .option("relationship.nodes.map", "false")
+      .option("relationship.source.labels", "Person")
+      .option("relationship.target.labels", "Person")
+      .load()
+
+    assertEquals(1, df.filter("`source.id` = 10 AND `target.id` = 1").collectAsList().size())
```

Comment:
the syntax is different from before where it was `<source>.<id>` ?
why? we should allow only one (simpler one)